### DISCUSSION TITLE
Can you drain connections from Kong upstream targets?

### DISCUSSION THREAD

**Discussion by blakqq** [Kong/kong#9154] [Help]
Hi there, hoping for some pointers in the right direction...

We have Kong deployed on a Kubernetes cluster, and use it as our egress to our data-centres from our micro-services (running in AWS EKS). When performing maintenance on one site, we simply remove that sites IP's as upstream targets from the Kong config-map, this causes a brief interruption for customers before connections are re-established using the opposite site.

Now that traffic is increasing, we'd prefer not to have such a brutal way of doing this, hoping for some tips on draining traffic from targets over a period of a few minutes, before then disabling or removing them?

I have been looking at the API, but it seems a lot of the features that look to be useful aren't available in DB-less mode, which being on a Kubernetes cluster, we are.

Any help or ideas would be appreciated. Thank you.

**Comment by chronolaw** [comment]
Could we  set the weight of the target to `0`?
